devito/data/allocators.py

```diff
@@ -317,6 +327,51 @@ def put_local(self):
         return self._node == 'local'
 
 
+class CupyAllocator(MemoryAllocator):
+
+    """
+    Memory allocator based on Unified Memory concept. The allocation is made using Cupy.
+    """
+    _mempool = None
+
+    @classmethod
+    def initialize(cls):
+
+        try:
+            import cupy as cp
```

Comment:
Move `try import` for cupy at the top

```
try:
  import cupy as cp
except importerror:
  cp = None
```

and just check if None here.

MPI should just be imported at the top from devito.mpi